charts/jaeger/templates/allInOne-config.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if .Values.allInOne.enabled -}}
```

Comment:
We can run jaeger-v2 binary without config and it will also work as all-in-one. Do we need this file? If people want to customize then can provide a custom config.

Reply:
yes we need it because if the user provides there own config file it will update this config template and then pass it to jaeger-v2 deployment. then how are we gonna decide that a user wanted to give there own config if we don't pass it or need to create a enabled flag which then again increase the things a user need to learn so we can simply pass it like this

Reply:
Don't follow your explanation. We must support the ability for the user to provide their custom config that completely replaces the default one. Given that as a requirements, why do we need a built-in "all in one" config in the chart if we already have it bundled with the binary directly? What use cases is that config meant to solve?

Reply:
well currently i am using this config map using 
```
  args:
             - "--config"
             - "/etc/jaeger/config.yaml"
```
if i decide not to use it i would have to add a extra if condition to this arg 
```
  args:
         {{- if values.config.enabled }}
             - "--config"
             - "/etc/jaeger/config.yaml"
         {{- end}}
```
this will unnecessary increase the flag plus we can use this config later for es and Cassandra  too.

Reply:
Once again, is it a REQUIREMENT that user can provide their own config file. How do you plan to support that?

Reply:
user can simply pass pass there config in eg . if he/she have a `test-config.yaml` file 
```
config:
  service:
    extensions: [jaeger_storage, jaeger_query, healthcheckv2]
    pipelines:
      traces:
        receivers: [otlp, jaeger, zipkin]
        processors: [batch]
        exporters: [jaeger_storage_exporter]

  extensions:
    healthcheckv2:
      use_v2: true
      http:
        endpoint: 0.0.0.0:13133
    jaeger_query:
      storage:
        traces: primary_store
        traces_archive: archive_store

    jaeger_storage:
      backends:
        primary_store:
          memory:
            max_traces: 90000
        archive_store:
          memory:
            max_traces: 90000

  receivers:
    otlp:
      protocols:
        grpc:
        http:
    jaeger:
      protocols:
        grpc:
    zipkin:

  processors:
    batch:

  exporters:
    jaeger_storage_exporter:
      trace_storage: primary_store
```
the user can pass this config file 
```
helm template charts/jaeger     --set provisionDataStore.cassandra=false     --set storage.type=memory     --set allInOne.enabled=true     --set agent.enabled=false     --set collector.enabled=false     --set query.enabled=false  -f config-test.yaml 
```

the output will look like this it will override the default values 
```
        extensions:
      
            healthcheckv2:
              http:
                endpoint: 0.0.0.0:13133
              use_v2: true
            jaeger_query:
              storage:
                traces: primary_store
                traces_archive: archive_store
            jaeger_storage:
              backends:
                archive_store:
                  memory:
                    max_traces: 90000
                primary_store:
                  memory:
                    max_traces: 90000
```

Reply:
what does `-f config-test.yaml ` mean? is this a helm flag? would it not override values.yaml completely, not just collector config?

Reply:
> if i decide not to use it i would have to add a extra if condition to this arg

no, you can use `allinone.enabled` as condition

Reply:
if allinOne is enabled then i deployment will always run and along with it the arguments will always be passed which means we will need a config map to pass with the arg